### ISSUE TITLE
std::variant constructor inheritance causes error squiggles

### ISSUE THREAD

**Issue by yaroslavsadin** [microsoft/vscode-cpptools#4636]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Windows 7 (Cygwin64, gcc 8.3.0)
- VS Code Version: 1.40.1
- C/C++ Extension Version: 0.26.2-insiders
- Other extensions you installed (and if the issue persists after disabling them):
- A clear and concise description of what the bug is.

While trying to use a class inherited from std:: variant and inherit its constructors, IntelliSense underlines this as an error. It compiles well though.

**To Reproduce**

Copy paste this code:

```
#include <variant>
#include <vector>

class D : public std::variant<int,std::vector<int>> {
    using std::variant<int,std::vector<int>>::variant;
};

int main(void) {
    std::vector<int> v {1,2,3,4};
    D d = D(v);
}
```

Compilation line: `g++ -std=c++17 inh_ctors.cpp`

Note error squiggle on line 10.

**Comment by sean-mcmanus** [comment]
Also repros on Linux. I think we need to isolate the code in the gcc headers used by `<variant>` to find the root cause.

**Comment by sean-mcmanus** [comment]
I've filed a bug on VS at https://developercommunity.visualstudio.com/content/problem/826530/cc-intellisense-incorrectly-reports-no-instance-of.html .

**Comment by sean-mcmanus** [comment]
@Colengms This bug still repros in VS Code, but I can't get a repro in VS -- any ideas how to get a VS repro?

**Comment by mmatrosov** [comment]
It is much simpler to reproduce it in VS code:

```
  void foo() {
    std::variant<int, double> var = 100;
  }
```

Gives this error:

> no suitable constructor exists to convert from "int" to "std::variant<int, double>"C/C++(415)

**Comment by sean-mcmanus** [comment]
Yeah, the root cause is probably https://developercommunity.visualstudio.com/content/problem/1278119/c-intellisense-gives-errors-with-the-variant-heade.html .

**Comment by sean-mcmanus** [comment]
@yaroslavsadin This issue appears fixed now.

@mmatrosov I'm not able to repro your issue. Can you file a new issue with more repro details? Such as what OS, compiler, etc. you're using?

**Comment by sean-mcmanus** [comment]
@yaroslavsadin @mmatrosov This was fixed for g++ but it still repros with clang, see https://developercommunity.visualstudio.com/content/problem/1278119/c-intellisense-gives-errors-with-the-variant-heade.html

**Comment by sean-mcmanus** [comment]
@yaroslavsadin We found a different IntelliSense bug that is with Cygwin only that can cause the system defines/includes to fail -- should be fixed with our pending 1.2.1 for next week.
